rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,185 @@
+
+The git histories of each branch will be maintained while merging. [Merging repositories](https://stackoverflow.com/questions/13040958/merge-two-git-repositories-without-breaking-file-history)
+
+### Webapps repository
+
+[repository](https://github.com/gravitational/webapps)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport repo](https://github.com/gravitational/teleport)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport/labels/ui) to
+        those transferred issues.
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PRs.
+  - [ ] Apply the necessary backports.
+
+### Webapps.e repository
+
+[repository](https://github.com/gravitational/webapps.e)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport.e repo](https://github.com/gravitational/teleport.e)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport.e/labels/ui)
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PR's.
+  - [ ] Apply the necessary backports.
+
+### Teleport repository
+
+[repository](https://github.com/gravitational/teleport)
+
+#### Actions
+
+- Remove `/webassets` submodule
+  - This submodule is no longer required as the web UI will be built on demand.
+- Clone the [Webapps repository](https://github.com/gravitational/webapps) into
+  the Teleport root. [Maintaining their respective git histories](https://stackoverflow.com/questions/13040958/merge-two-git-repositories-without-breaking-file-history)
+  - This will need to be done for every respective version branch (v9, v10, v11)
+- update targets that use the `packages/webapps.e` submodule to point points to
+  the correct version in the `teleport.e/webapps` folder.
+- only require teleport build processes to run on teleport paths and the webapp
+  ones to run on the webapp paths
+-
+- Archive the [Webapps repository](https://github.com/gravitational/webapps).
+
+#### CI jobs
+
+- **CodeQL / Analyze (go) (pull_request)**
+- **Teleport-IntegrationTest (ci-account)**
+- **Teleport-UnitTest (ci-account)**
+- **Teleport-DocTest (ci-account)**
+- **Teleport-Lint (ci-account)**
+- **Teleport-OsCompatibility (ci-account)**
+  - Disable for changes exclusively made in the `/webapp` path.
+- **Assign / Auto Request Review**
+  - Create a list of full stack engineers and have it pick from this list for
+    changes made to the `/webapp` path.
+- **Check / Checking reviewers**
+  - Continue to require 2 reviewers for changes in the `webapp` projects.
```

Comment:
This unfortunately is not up for debate. We must enforce two reviewers for our SOC-2.

We got away with not enforcing 2 reviewers on webapps in the past because we required 2 approvals to update the submodule and incorporate the changes into teleport. Now that all the code will live in teleport, we'll be leveraging the same review process and requirements as any other teleport change.

Reply:
> This unfortunately is not up for debate. We must enforce two reviewers for our SOC-2.

Yes, I understand that. What I meant is not that we should maybe allow for just one reviewer but rather that in the past you had at least devs looking at your backend code and at least one fullstack looking at the frontend. Ending up with two backend devs looking at frontend changes or one backend and one fullstack dev feels just slightly worse in comparison.

Reply:
> Ending up with two backend devs looking at frontend changes or one backend and one fullstack dev feels just slightly worse in comparison.

We are able to add additional requirements to the reviewer script so that changes to the `webapps` folder use a different list of developers. I can expand this line item to include that script change as well.